modules/system/base.nix

```diff
@@ -5,6 +5,14 @@
     IFS="." read -r -a macOSVersion <<< "$(sw_vers -productVersion)"
 
     if [[ ''${macOSVersion[0]} -gt 10 || ( ''${macOSVersion[0]} -eq 10 && ''${macOSVersion[1]} -ge 15 ) ]]; then
+      if [[ $(stat -c '%a' /etc/synthetic.conf) != "644" ]]; then
```

Comment:
One thing that was confusing to me here is I originally wrote this for macOS's `/usr/bin/stat` which has a different argument format. It appears that the environment that this runs in uses a gnu `stat` but please check me here.

Reply:
Yeah fairly sure `stat` comes from `coreutils` and the command works for me:

```
$ nix shell nixpkgs#coreutil -c stat -c '%a' /etc/synthetic.conf
600
```

Reply:
Ahh that makes total sense, thanks for clarifying!